Use apk package-managed installation of php7-gd

jayaddison opened the pull request:
Although it requires a couple of other small changes as a side-effect, we can remove the requirement for code compilation at `grocy` container build-time by using a version of [`php7-gd` from `apk`](https://pkgs.alpinelinux.org/package/v3.11/community/x86/php7-gd).

Resolves #50 
Resolves #69

jayaddison commented:
@tet3 Thanks for the help with documentation in #68.  If you have time, could you test out a local `docker-compose build` of this branch?  It should simplify some of the PHP dependency management for the `grocy` application container.

tet3 commented:
@jayaddison Is #69 the reason you mentioned me here? This branch resolves that issue, as far as I can tell. In production mode, the problematic `ENTRY_PAGE` values all work, and manually editing the URL to one of those values redirects to login when not logged in.

jayaddison commented:
@tet3 That's partly the reason, although this is also a large-ish change and you seem like an experienced user, so your testing feedback is valuable.

I haven't yet been able to figure out *why* this PR resolves the login redirect issue in #69 - and I'd definitely be curious to find out.

jayaddison commented on docker_grocy/www.conf:
```diff
@@ -0,0 +1,6 @@
+daemonize = no
+
+[www]
+clear_env = no
```
@tet3 This is one potential candidate.  I don't love this change, since it potentially enables a whole bunch of environment variables to pass from the host to the PHP application.

If we could find a way to selectively allow only `GROCY_`-prefixed variables (i.e. genuine application variables) to the application, that could be an improvement.  Or perhaps there should be a larger design re-think here.

jayaddison replied:
Writing down these thoughts in a more structured form, to approximate requirements:

- The app's (non-default?) configured environment values should be committed to source control in this repository, so that they're easily visible and modified
- Only environment variables relevant to the application should be made available to the container(s) from the host
- It should be possible for end-users of the Docker images to override any of the application's environment variables at runtime -- without rebuilding the image
- Sensible default values should be provided in released Docker images so that new users can get started with only a command or two

jayaddison commented:
**Image Size Comparison**

Before:
```
grocy-docker $ git checkout master
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
grocy-docker $ git rev-parse --short HEAD 
5fbb7f0
grocy-docker $ GROCY_VERSION=v2.6.1 docker-compose build > /dev/null
Building grocy
Building grocy-nginx
grocy-docker $ docker images --filter reference=grocy/grocy-docker
REPOSITORY           TAG                 IMAGE ID            CREATED             SIZE
grocy/grocy-docker   nginx               a0275cd09a33        1 second ago        9.24MB
grocy/grocy-docker   grocy               753bb3bd6af3        5 seconds ago       239MB
```

After:
```
grocy-docker $ git checkout apk-managed-php-gd 
Switched to branch 'apk-managed-php-gd'
grocy-docker $ git rev-parse --short HEAD 
d2e2891
grocy-docker $ GROCY_VERSION=v2.6.1 docker-compose build > /dev/null
Building grocy
Building grocy-nginx
grocy-docker $ docker images --filter reference=grocy/grocy-docker
REPOSITORY           TAG                 IMAGE ID            CREATED             SIZE
grocy/grocy-docker   nginx               74b3e8d4a2ce        30 seconds ago      9.24MB
grocy/grocy-docker   grocy               b944bd08aab2        35 seconds ago      188MB
```
